[wip] migration tool backend

otan opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76135)
<!-- Reviewable:end -->

otan committed:
initial commit of migration server

Release note (<category, see below>): <what> <show> <why>

Release justification:

otan committed:
initial import

otan committed:
add web

Release note (<category, see below>): <what> <show> <why>
